src/gui/C4StartupScenSelDlg.cpp

```diff
@@ -1442,6 +1442,21 @@ C4StartupScenSelDlg::C4StartupScenSelDlg(bool fNetwork) : C4StartupDlg(LoadResSt
 	pScenSelCaption = new C4GUI::Label("", caBookLeft.GetFromTop(rScenSelCaptionFont.GetLineHeight()), ACenter, ClrScenarioItem, &rScenSelCaptionFont, false);
 	pSheetBook->AddElement(pScenSelCaption);
 	pScenSelCaption->SetToolTip(LoadResStr("IDS_DLGTIP_SELECTSCENARIO"));
+
+	// search bar
+	const char *labelText = LoadResStr("IDS_DLG_SEARCH");
+	int32_t width = 100, height;
```

Comment:
Height is either unused, or initialized by GetTextExtent? Anyway, please define variables in separate lines, and add a comment why they are not initialized, but used in later lines.

Reply:
>Anyway, please define variables in separate lines
Why? [The style guidelines don't forbid it.](https://wiki.openclonk.org/w/Style_Guidelines#Data_Types)

Reply:
I updated the style guidelines :D

The reason is easy, though: Although the code compiles, I do not know whether:
- You forgot assigning a value.
- You declared a variable but do not use it (especially applies if there are multiple in a line).
These are things that a code analyzer can easily tell you, but I do not have that when I am looking just at the code. Having the variable declarations in separate lines will point your attention towards them, especially unassigned ones.

In the updated guidelines I decided that multiple declarations of the same type within a line are OK, though, if no value is assigned. Any opinions on that?

Reply:
I'd rather have everything assigned a value. The compiler can eliminate dead stores. GetTextExtent should probably return a std::pair (or maybe a new Dimension type).

Reply:
An additional (classical) example to what Marky said:
`C4Gui::ListBox* left, right;` would declare one pointer and one object on the stack.

In general, we do not (and can't) cover everything with the style guidelines. The guidelines are mainly for the things people usually argue about (e.g. tabs vs. commas, variable names...).